tensordict/tensordict.py

```diff
@@ -1669,6 +1688,12 @@ def unflatten_keys(
             out = self
 
         for key, list_of_keys in to_unflatten.items():
+
+            if key in out.to_dict():
```

Comment:
isnt `out` already a TD? If yes, we can check for membership without mapping it back to `dict` no ?

Reply:
Indeed, `to_dict` is an expensive operation. I see there is one here and one above. 
In this case I would simply do
```python
        keys = set(self. Keys()) # pre-compute the keys
        for key, list_of_keys in to_unflatten.items():

            if key in keys:
                ...
```

Reply:
Curious if even `keys = set(self. Keys())` is also needed, as keys in the dict are unique

Reply:
changed `.to_dict()` to `.keys()` in both of these instances

Reply:
> Curious if even `keys = set(self. Keys())` is also needed, as keys in the dict are unique

It's not for the uniqueness but because querying the key does not come for free and we don't want to do it several times in a loop. A set is preferable than a list as membership is faster to compute (I think?)

Reply:
> It's not for the uniqueness but because querying the key does not come for free and we don't want to do it several times in a loop. A set is preferable than a list as membership is faster to compute (I think?)

makes sense, I'll update this